sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/training/CustomModelTransforms.java

```diff
@@ -54,55 +57,147 @@ static CustomFormModel toCustomFormModel(Model modelResponse) {
-            Map<String, CustomFormModelField> fieldMap = new TreeMap<>();
-            modelResponse.getKeys().getClusters().forEach((clusterKey, clusterFields) -> {
-                forEachWithIndex(clusterFields, (index, eachField) -> {
-                    String fieldName = "field-" + index;
-                    fieldMap.put(fieldName, new CustomFormModelField(eachField, fieldName, null));
-                });
-                subModelList.add(new CustomFormSubmodel(
-                    null,
-                    fieldMap,
-                    formType + clusterKey));
-            });
+            // unlabeled model, read from page results
+            subModelList = getUnlabeledSubmodels(modelResponse.getKeys().getClusters(), modelId);
         } else if (modelResponse.getTrainResult() != null && modelResponse.getTrainResult().getFields() != null) {
             // labeled model
-            Map<String, CustomFormModelField> fieldMap = new TreeMap<>();
-            modelResponse.getTrainResult().getFields()
-                .forEach(formFieldsReport -> fieldMap.put(formFieldsReport.getFieldName(),
-                    new CustomFormModelField(null, formFieldsReport.getFieldName(),
-                        formFieldsReport.getAccuracy())));
-            subModelList.add(new CustomFormSubmodel(
-                modelResponse.getTrainResult().getAverageModelAccuracy(),
-                fieldMap,
-                formType + modelInfo.getModelId()));
+            String formType = "custom:";
+            if (modelInfo.getModelName() != null) {
+                formType = formType + modelInfo.getModelName();
+            } else {
+                formType = formType + modelInfo.getModelId();
+            }
+            subModelList = getLabeledSubmodels(modelResponse, modelId, formType);
+        } else if (!CoreUtils.isNullOrEmpty(modelResponse.getComposedTrainResults())) {
+            // composed model
+            subModelList = getComposedSubmodels(modelResponse);
+            trainingDocumentInfoList = new ArrayList<>();
+            for (TrainResult composedTrainResultItem : modelResponse.getComposedTrainResults()) {
+                final List<TrainingDocumentInfo> trainingDocumentSubModelList
+                    = getTrainingDocumentList(composedTrainResultItem.getTrainingDocuments(),
+                    composedTrainResultItem.getModelId().toString());
+                trainingDocumentInfoList.addAll(trainingDocumentSubModelList);
+            }
         }
 
-        return new CustomFormModel(
+        CustomFormModel customFormModel = new CustomFormModel(
             modelInfo.getModelId().toString(),
             CustomFormModelStatus.fromString(modelInfo.getStatus().toString()),
             modelInfo.getCreatedDateTime(),
             modelInfo.getLastUpdatedDateTime(),
             subModelList,
             modelErrors,
             trainingDocumentInfoList);
+
+        if (modelInfo.getAttributes() != null) {
```

Comment:
what happens if the service doesn't return the attributes? 
i.e. for a model trained without labels, the service won't return this. 
Will the user be able to ask if `model.properties.isComposed` ?

Reply:
No, in the case of model trained without labels, the `model.properties` would be null and so the user won't be able to drill down further to the `isComposed` property.

Reply:
It will be better for the user to have a consistent experience when looking into the `CustomFormModel` independent of model trained with labels or not. If the user has a list of models and wants to see if they are composed or not they will have to first check if `Properties` are null and if not then check if `isComposed == true`.
It will be better for the SDK to always initialized `Properties` with `isComposed=false` so that the user can always count on `Properties` being present and therefore just as `isComposed`.

Reply:
We are doing this in .NET and Python

Reply:
I think it's a general pattern we have used in Java libraries to keep properties as null rather than using some kind of defaulting. 

Also, to think about it wouldn't `isComposed=false` mean that the model is not composed but isComposed/properties as null would mean it cannot have such a property which is the case for "models trained without labels".

I don't think `isComposed=false` is particularly true for models trained without labels. `null` and `false` would mean different things following Java standards.

Reply:
This might be language specific then. In .NET we avoid null as much as possible as it breaks the flow for customers.

I still think you should initialize it, otherwise you are saying that `Properties` only applied for models with labels.
i.e. if in the future the service adds a new property that applies for both models with and without labels, you are going to have to initialize `Properties` and assigne `isComposed` to false. This will already brake customers that are assuming that `Properties` are only for models with labels.

Reply:
> if in the future the service adds a new property that applies for both models with and without labels, you are going to have to initialize Properties and assigne isComposed to false.

Not true. In that case, too, the properties would have to be assigned but `isComposed` should only be assigned with what's returned from the service (true/null/false).  

Defaulting when service doesn't return values could be a considerable overhead for client libraries.  For instance, when the service adds additional 5-6 properties, we would be defaulting them all in case of .NET approach?

Reply:
So should Properties be assigned now? will it be confusing if later properties is never null?
It depends on the property, so we default case by case. Currently we do default to false for `isComposed`

Reply:
Asked the service => https://teams.microsoft.com/l/message/19:88f2d9dea2344075b5d4bc34d82a3d1c@thread.skype/1602801673358?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&parentMessageId=1602801673358&teamName=Azure%20SDK&channelName=Service%20-%20Form%20Recognizer&createdTime=1602801673358